typedb/api/connection/options.py

```diff
@@ -46,6 +47,42 @@ def cluster() -> "TypeDBClusterOptions":
     def is_cluster(self) -> bool:
         return False
 
+    def set_infer(self, infer: bool):
+        self.infer = infer
+        return self
+
+    def set_trace_inference(self, trace_inference: bool):
+        self.trace_inference = trace_inference
+        return self
+
+    def set_explain(self, explain: bool):
+        self.explain = explain
+        return self
+
+    def set_parallel(self, parallel: bool):
+        self.parallel = parallel
+        return self
+
+    def set_prefetch_size(self, prefetch_size: int):
+        self.prefetch_size = prefetch_size
+        return self
+
+    def set_prefetch(self, prefetch: bool):
+        self.prefetch = prefetch
+        return self
+
+    def set_session_idle_timeout_millis(self, session_idle_timeout_millis: int):
+        self.session_idle_timeout_millis = session_idle_timeout_millis
+        return self
+
+    def set_transaction_timeout_millis(self, transaction_timeout_millis: int):
+        self.transaction_timeout_millis = transaction_timeout_millis
+        return self
+
+    def set_schema_lock_acquire_timeout_millis(self, schema_lock_acquire_timeout_millis: int):
+        self.schema_lock_acquire_timeout_millis = schema_lock_acquire_timeout_millis
+        return self
```

Comment:
implement fluent interface like in java to configure options, mostly to allow simple lambdas to configure options in BDD, but also a useful API to have

Reply:
since we have setters, we also add getters